proposed/2021/TransitiveDependenciesInPMUI.md

```diff
@@ -0,0 +1,95 @@
+* Issue: [10184](https://github.com/NuGet/Home/issues/10184) Show transitive dependencies in PMUI for PackageReference projects
+* Type: Feature
+
+## Problem Background
+
+Developers can only see their top-level packages in their PackageReference project in the Package Manager UI. Transitive dependencies are not visible in the NuGet Package Manager. This makes it difficult for a developer to fully understand all of the NuGet package dependencies their project relies on. Furthermore, there is currently no way to see in the Package Manager UI what top-level dependency depends on each of the included transitive dependencies.
+
+A developer may want to install a transitive dependency as a top-level one to specify what version they want to include in their project if there is a security concern, vulnerability, or version conflict with the dependant version of a transitive dependency. The developer should have a quick and easy way to do this from the Package Manager UI in Visual Studio.
+
+## Who are the customers
+
+* All developers using Visual Studio and PackageReference projects will benefit from the additional information about the transitive dependencies and the top-level package that depend on them.
+* Developers that have installed packages that depend on transitive dependencies with security concerns, vulnerabilities, or version conflicts with other transitive dependencies will benefit from the ability to quickly install a transitive dependency as a top-level dependency and specify its version.
+
+## Requirements
+
+* UI is accessible and allows screen readers to announce pertinent information about the transitive dependencies and the top-level packages that depend on them
+* UI is localizable. All strings are in appropriate resource files and available for translation
+* Reading transitive package information is done asynchronously to minimize the impact on the responsiveness of the UI
+
+## Goals
+
+* Show transitive dependencies in the Project-level NuGet Package Manager
+
+### UI behavior goals:
+* Display a flat list of transitive depedencies along with the installed, top-level packages in the Visual Studio NuGet Package Manager UI
+* Allow the user to collapse and expand the list of transitive dependencies
+* Collapse the transitive dependencies list by default
+* For each transitive dependency, show the top-level package(s) that depend(s) on it in a tooltip, displayable on hover and readable by screen reader
+* Show the package details when a transitive dependency is selected and allow the ability to install a specific version of the transitive dependency which will promote it to a top-level package
+
+### Technical goals:
+* Enhance the InfiniteScrollList to support grouping its list content based on a field in the list item data context
+* Allow the InfiniteScrollList grouping to be enabled or disabled by the consumer so it can still be shared by the Browse, Installed, Updates tabs
```

Comment:
> so it can still be shared by the Browse, Installed, Updates tabs

The first part of the bullet I thought was about giving a checkbox/etc to enable/disable grouping.
This statement is confusing. What is being shared? If you mean the user's choice to enable/disable, are we certain that's what we want? Previous discussions with PM it sounded like some settings should be per-tab, not shared.

Reply:
The "it" in this sentence refers to the InfiniteScrollList. This design adds support for the InfiniteScrollList to be grouped optionally and I'm emphasizing that it will still support displaying its contents without grouping. This is important so that the same control can still be used in all tabs. The optionality here is programmatic and not driven by user action. We will not be exposing any option to allow for disabling grouping because that doesn't fit into the design.

Reply:
> We will not be exposing any option to allow for disabling grouping because that doesn't fit into the design.

Ah ok, I wasn't aware of that change. @JonDouglas does this sound like what you expected?

Reply:
The [design spec](https://github.com/NuGet/Home/blob/dev/proposed/2020/Transitive-Dependencies.md) outlines how the UI should behave. Grouping the list by top-level versus transitive packages is at the core of the design as the way to make the difference between the packages (top-level vs transitive) easily distinguishable. Allowing the user to ungroup them has never been a consideration that I am aware of.

Reply:
Nevermind, since the "it" meant a specific control, I no longer have a question about a shared user-setting.